keep/providers/splunk_provider/splunk_provider.py

```diff
@@ -0,0 +1,154 @@
+        },
+        default=8089
+    )
+
+
+class SplunkProvider(BaseProvider):
+    """Pull alerts and query incidents from Splunk."""
+
+    PROVIDER_SCOPES = [
+        ProviderScope(
+            name="connect_to_client",
+            description="The user can connect to the client",
+            mandatory=True,
+            alias="Connect to the client",
+        )
+    ]
+
+    SEVERITIES_MAP = {
+        "1": AlertSeverity.LOW,
+        "2": AlertSeverity.INFO,
+        "3": AlertSeverity.WARNING,
+        "4": AlertSeverity.HIGH,
+        "5": AlertSeverity.CRITICAL,
+    }
+
+    def __init__(
+            self, context_manager: ContextManager, provider_id: str, config: ProviderConfig
+    ):
+        super().__init__(context_manager, provider_id, config)
+
+    def validate_scopes(self) -> dict[str, bool | str]:
+        try:
+            connect(username=self.authentication_config.username, password=self.authentication_config.password,
+                    host=self.authentication_config.host, port=self.authentication_config.port)
+            scopes = {
+                "connect_to_client": True,
+            }
+        except Exception as e:
+            self.logger.exception("Error validating scopes")
+            scopes = {
+                "connect_to_client": str(e),
+            }
+        return scopes
+
+    def validate_config(self):
+        self.authentication_config = SplunkProviderAuthConfig(
+            **self.config.authentication
+        )
+
+    def dispose(self):
+        """
+        No need to dispose of anything, so just do nothing.
+        """
+        pass
+
+    def setup_webhook(
+            self, tenant_id: str, keep_api_url: str, api_key: str, setup_alerts: bool = True
+    ):
+        self.logger.info("Setting up Splunk webhook on all Alerts")
+        creation_updation_kwargs = {
+            "actions": "webhook",
+            "action.webhook": "1",
+            "action.webhook.param.url": keep_api_url,
+        }
+        service = connect(username=self.authentication_config.username, password=self.authentication_config.password,
+                          host=self.authentication_config.host, port=self.authentication_config.port)
+        for saved_search in service.saved_searches:
+            existing_webhook_url = saved_search["_state"]["content"].get("action.webhook.param.url", None)
+            if existing_webhook_url is None or existing_webhook_url != keep_api_url:
+                print("REEE_LOGS_HERE: ", saved_search["path"], existing_webhook_url)
+                saved_search.update(**creation_updation_kwargs).refresh()
+
+    # @staticmethod
+    def _format_alert(self, event: dict) -> AlertDto:
+        search_id = event["sid"]
+        service = connect(username=self.authentication_config.username, password=self.authentication_config.password,
+                          host=self.authentication_config.host, port=self.authentication_config.port)
```

Comment:
You won't have all of these when getting events on the fly (This method is static for that reason)
We need to think about it more carefully (perhaps inject an initialized Provider instance when the alerts comes in from an existing provider)

keep_api_url that you used in the setup_webhook basically includes the provider id so we can maybe do that

cc: @shahargl

Reply:
@talboren is the datadog provider working fine??

Reply:
![image](https://github.com/keephq/keep/assets/70096901/6446cf81-3941-47d3-a2ba-5610bd0fb759)
it too dosen't implement a `@staticmethod`

Reply:
It is decorated with @staticmethod in the parent `BaseProvider`, it's just for IDE to note that it is staticmethod

Reply:
resolved in https://github.com/keephq/keep/pull/950/commits/ec908657343359472bdffc6a150771e41fd56e65